aperturedb/secrets.go

```diff
@@ -0,0 +1,171 @@
+package aperturedb
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"database/sql"
+	"fmt"
+
+	"github.com/lightninglabs/aperture/aperturedb/sqlc"
+	"github.com/lightninglabs/aperture/lsat"
+	"github.com/lightninglabs/aperture/mint"
+	"github.com/lightningnetwork/lnd/clock"
+)
+
+type (
+	// NewSecret is a struct that contains the parameters required to insert
+	// a new secret into the database.
+	NewSecret = sqlc.InsertSecretParams
+)
+
+// SecretsDB is an interface that defines the set of operations that can be
+// executed against the secrets database.
+type SecretsDB interface {
+	// InsertSecret inserts a new secret into the database.
+	InsertSecret(ctx context.Context, arg NewSecret) (int32, error)
+
+	// GetSecretByHash returns the secret that corresponds to the given
+	// hash.
+	GetSecretByHash(ctx context.Context, hash []byte) ([]byte, error)
+
+	// DeleteSecretByHash removes the secret that corresponds to the given
+	// hash.
+	DeleteSecretByHash(ctx context.Context, hash []byte) (int64, error)
```

Comment:
Do we actually need the returned primary key here? Or seems like a newer `sqlc` feature maybe?

Reply:
this does not return the primary key but the number of rows affected by this query. The idea is to be able to return an error if we are deleting a hash that is not currently in the db